src/util/fetch-with-timeout.js

```diff
@@ -1,3 +1,25 @@
+/**
+ * @callback FetchFunction
+ * @param {RequestInfo|URL} input
+ * @param {RequestInit|undefined} [init]
+ * @returns {Promise<Response>}
+ */
+
+/**
+ * @type {FetchFunction}
+ */
+let myFetch = global.fetch;
+
+/**
+ * Tell `fetchWithTimeout` to use a specific `fetch` function.
+ * By default, `fetchWithTimeout` will use the global `fetch` function.
+ * If there is no global `fetch`, then `fetchWithTimeout` will fail unless provided with an alternative.
+ * @param {FetchFunction} newFetch The new `fetch` function to use within fetchWithTimeout.
+ */
+const setFetch = newFetch => {
```

Comment:
Why do we need to set a fetch function? What cases are we not using the original fetch method for? And should it be a consistent way to make fetch requests across the codebase?

Reply:
See above, but also...

> And should it be a consistent way to make fetch requests across the codebase?

Yep! We're starting with a few specific things, but we intend to eventually use this for all HTTP requests in the Scratch editor.

Reply:
That makes sense, thanks for clarifying!